generator/src/test/java/com/scottlogic/deg/generator/cucumber/LessThan.feature

```diff
@@ -2,12 +2,332 @@ Feature: User can specify that a numeric value is lower than, but not equal to,
+Scenario: User requires negative numbers data
+     Given foo is less than 0
+       And the generator can generate at most 5 rows
+     Then the following data should be included in what is generated:
+       | foo |
+       | -1  |
+
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: User requires 0
+     Given foo is less than 1
+       And the generator can generate at most 5 rows
+     Then the following data should be included in what is generated:
+       | foo |
+       | 0   |
+
+#lessThan
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a non contradicting lessThan should be successful (lessThan 2 AND lessThan 5)
+     Given foo is less than 2
+       And foo is less than 5
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
+
+Scenario: lessThan run against a non contradicting not lessThan should be successful (lessThan 5 AND not lessThan 1)
+     Given foo is less than 5
+       And foo is anything but less than 1
+     Then the following data should be generated:
+       | foo |
+       | 1   |
+       | 2   |
+       | 3   |
+       | 4   |
+
+Scenario: not lessThan run against a non contradicting lessThan should be successful (not lessThan 2 AND lessThan 5)
+     Given foo is anything but less than 2
+       And foo is less than 5
+     Then the following data should be generated:
+       | foo |
+       | 2   |
+       | 3   |
+       | 4   |
+
+Scenario: not lessThan run against a non contradicting not lessThan should be successful (not lessThan 5 AND not lessThan 5)
+     Given foo is anything but less than 5
+       And foo is anything but less than 5
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo |
+       | 5   |
+       | 6   |
+       | 7   |
+       | 8   |
+       | 9   |
+
+Scenario: lessThan run against a contradicting not lessThan should only only generate string, temporal and null (lessThan 2 AND not lessThan 2)
+     Given foo is less than 2
+       And foo is anything but less than 2
+     Then no data is created
+
+#lessThanOrEqual
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a non contradicting lessThanOrEqualTo should be successful (lessThan 6 AND lessThanOrEqualTo 5)
+     Given foo is less than 6
+       And foo is less than or equal to 5
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
```

Comment:
Unsure of these expected results. Would have expected <6 and <=5 to create:
5
4
3
2
1

Reply:
Ordering again, they would be I think but would be last

Reply:
Ref #594